src/Driver/Mysqli/Connection.php

```diff
@@ -93,7 +93,13 @@ public function exec(string $sql): int
             throw ConnectionError::new($this->connection);
         }
 
-        return $this->connection->affected_rows;
+        $count = $this->connection->affected_rows;
+
+        if (is_string($count)) {
+            $count = PHP_INT_MAX;
```

Comment:
This is by no means acceptable. The library must not alter the data returned by the underlying driver in order to fit it into its own API which is incorrectly defined.

Reply:
What would be the alternative? Right now, if the number of affected rows if larger than `PHP_INT_MAX` and thus returned as a string by the native driver, we will run into a `TypeError` here.

https://3v4l.org/03d4D

Reply:
The return type should be `int|string` as it is in `4.0.x`:

https://github.com/doctrine/dbal/blob/8622d1d4c9e4d44e4f1db0111cd9390e57cbfdf9/src/Driver/Mysqli/Connection.php#L60

Reply:
Understood, but don't we need a solution for DBAL 3 as well? After all, this is our maintained branch. Raising type errors seems like a mistake to me.

Reply:
As discussed in https://github.com/doctrine/dbal/pull/5317#issuecomment-1076826375, it seems acceptable to remove the `int` return type declaration in `3.0.x` since it's invalid.

Reply:
Okay, let's do that.